### ISSUE TITLE
config.lua should be added to gitignore

### ISSUE THREAD

**Issue by Dublerq** [otland/forgottenserver#1772]
### Expected behaviour
In my opinion config.lua should be added to .gitignore. Someone may accidentally push his config to repository...

### Actual behaviour
It's not

**Comment by ranisalt** [comment]
I disagree, but we should rather ship it as `config.lua.sample` or other name that means it's just an example and should be modified, then add to .gitignore.

**Comment by Dublerq** [comment]
@ranisalt That's what I ment

**Comment by daniofordon** [comment]
I am also of the opinion

**Comment by ranisalt** [comment]
@PrinterLUA go ahead and PR :P

**Comment by ranisalt** [comment]
Wtf I don't know why I mentioned PrinterLUA. I meant @Dublerq but just saw now.

**Comment by Shawak** [comment]
Why not simply add a `dofile('../config.lua')` at the add of the config file and overwrite settings like the database password, so even if you push your config by a mistake it won't be a huge one.

**Comment by ranisalt** [comment]
@Shawak because the default config.lua is **meant** to be read and modified. By forcing users to rename it they at least take note and we prevent any leaking of info anyway.

**Comment by Codex-NG** [comment]
Why is config a lua file anyway, wouldn't it be simpler to parse it as xml, this would also simplify parsing it in the AAC's?

**Comment by ranisalt** [comment]
> Why is config a lua file anyway, wouldn't it be simpler to parse it as xml, this would also simplify parsing it in the AAC's?

Because you can use functions to define values. Lua is powerful as a programming language while XML is only for markup, crude values. Yet I never seen no one use it like that, you can e.g. have a function to apply double exp on weekends.

AACs ideally should not parse the Lua file, but rather communicate via the status protocol, that returns XML data.

**Comment by Codex-NG** [comment]
You can still assign values in the same manner in lua even if the config file is in an xml format, these functions which call on the values from config are calling them from server not the config file directly and even if you think its ugly its still much simpler to do and frankly more organized.

**Comment by ranisalt** [comment]
No, not really. The Lua engine is called to run the config.lua file, so you can use any function available on the Lua core. With a little twist you can even do HTTP stuff on config.lua e.g. check [icanhazip.com](icanhazip.com) to set the IP automatically. It would not be possible with XML.

**Comment by Codex-NG** [comment]
Yes, yes really, if this wasn't the case then we would not need `ConfigManager` when referencing something from config.lua, btw you are linking to a page which no longer exists.

xml is more user friendly when it comes to data manipulation, I've been writing scripts in lua for about 2 or 3 years, but never gave c++ a chance now I am seeing that we don't even need lua, if anything lua seems more like a burden on the server than a helpful extension of the server's functionality.

That is just my opinion, to each their own.

**Comment by gugahoa** [comment]
www.icanhazip.com 
Lua is easy to learn and easy to use. The engine provides a safe environment so that a scripter doesn't need to know what is memory allocation, what's dangling pointer, what's free, what's what.
XML stands for EXtensible Markup Language. It's a Markup Language. Lua is a scriptwriting language, that's the difference that makes it better to use Lua instead of XML. Ever heard of someone using XML for scriptwriting? It's ok to use XML to represent data, but I don't see how it can prevail over Lua for scriptwriting.

config.lua is in Lua because of how powerful Lua can be for configuration. As @ranisalt mentioned, you could automate lots of things with Lua with the simple addition of one `variable = statement and statement or statement`

**Comment by Codex-NG** [comment]
You don't need to explain what lua, c++ or xml is, I know what they are and their capabilities. I just feel lua is used extremely too much, 99% of the concepts I've written since I started editing the sources don't even need lua to function.. The things I am building or have built already would take me 100's of lines of code across several scripts in lua while using an excessive amount of memory/cpu usage, where as in c++ is a tiny fraction of that with little to almost no memory/cpu usage. 

How?
It's all in the design :)

**Comment by ranisalt** [comment]
Man, you definitely don't know what you're talking about.

**Comment by Codex-NG** [comment]
@ranisalt you don't know what you are talking about, I don't think you ever have, all you do is release broken code. :(

**Comment by Dublerq** [comment]
I'd be suprised if you'd need ANY lua function to hardcode something in C++ written server. Lua scripting is much faster and the code is easy to reuse. That's why even advanced programmers (7y experience with OTS) choose it over C++. I love seeing that you love newly learned C++ but still you need to learn which tool and when you should use complete your goals.

**Comment by andersonfaaria** [comment]
Codex, you're really that stupid or are you just forcing? There's a reason behind the use of script languages in an application. It's not the first time I've seen you talking bullshit in healthy discussions, if you don't know about something, go google it instead of spreading your newbie opinions as the supreme truth. Sincerely yours, xWhiteWolf

**Comment by Codex-NG** [comment]
@andersonfaaria you must be a procrastinator if you waited all this time to speak up about my conduct on otland, since this is the case I will take what you have to say with a grain of salt.

The name `xWhiteWolf` does not ring any bells, which means you have not made in impact in my life or anyone else's for that matter.. nothing new there.

Whether I am loved or hated by you people, (loved when I was your coding slave) and now hated because I tell you all you need to stop copy and pasting code and go to school or read a book and also because I won't be your coding slave... 

To summarize: I don't care what any of you think about me or what you think you know, I have more knowledge in my left pinky than 90% of you have collectively on otland.

Good day.

**Comment by Dublerq** [comment]
There is no need to discuss that topic here. That's not the point of github issues.

**Comment by Tarek1337** [comment]
@Codex-NG ok, pls come back coding slave we need u

**Comment by marksamman** [comment]
Please refrain from writing comments that don't contribute to solving the issue. I've deleted a few off-topic/offensive messages and blocked @Codex-NG from the repository.

I've shared my opinion here: https://github.com/otland/forgottenserver/pull/1888#issuecomment-242503897

**Comment by soul4soul** [comment]
@DSpeichert 
This issue has been closed with commit 20df00b
